nrf-hal-common/src/pwm.rs

```diff
@@ -973,22 +1090,56 @@ pub enum Error {
     BufferTooLong,
 }
 
-pub trait Instance: private::Sealed {}
+pub trait Instance: private::Sealed + Deref<Target = crate::pac::pwm0::RegisterBlock> {
+    const INTERRUPT: Interrupt;
 
-impl Instance for PWM0 {}
+    /// Provides access to the associated internal duty buffer for the instance.
+    fn buffer() -> &'static mut Cell<[u16; 4]>;
```

Comment:
Using this signature will create aliasing mutable references when called twice, can we return an immutable ref instead? Since it's a `Cell` you can still use interior mutability.

Reply:
Good idea!